File: 2.Algorithm/99.ETC/영어 끝말 잇기/영어 끝말 잇기.py
def solution(n, words):
    first = False
    for i in range(1,len(words)):
        if words.count(words[i]) >= 2:
            if first or i == len(words)-1:
                return [(i%n)+1,(i//n)+1]
            else:
                first = True
        elif words[i-1][-1:] != words[i][0]:
            return [(i%n)+1,(i//n)+1]
    return [0,0]

File: 2.Algorithm/99.ETC/영어 끝말 잇기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int n, String[] words) {
        boolean first = false;
        for (int i = 1; i < words.length; i++){
            String s = words[i];
            if (Arrays.stream(words).filter(x -> x.equals(s)).count() >= 2){
                if(first || i == words.length-1){
                    return new int[] {i % n + 1, i / n + 1};
                }else{
                    first = true;
                }
            }else if(words[i-1].charAt(words[i-1].length()-1) != words[i].charAt(0)){
                return new int[] {i % n + 1, i / n + 1};
            }
        }
        return new int[] {0, 0};
    }
}